content/tutorials/map_algebra/basic_map_algebra.qmd

```diff
@@ -0,0 +1,585 @@
+```{bash}
+r.surf.fractal output=fractal dimension=2.25
+r.info -r map=fractal format=plain
+```
+## Python
+```{python}
+# Generate fractal surface
+gs.run_command("r.surf.fractal", output="fractal", dimension=2.25)
+
+# Visualize
+m.d_rast(map="fractal")
+m.d_legend(raster="fractal", at=(5, 95, 1, 3))
+m.show()
+
+# Print range
+info = gs.read_command("r.info", map="fractal", flags="r")
+print(info)
+```
+:::
+
+![Fractal](images/map_algebra_04.webp)
+
+# Arithmetic
+
+## Operations
+Next let's try arithmetic with map algebra.
+Since our fractal mountains are too high, 
+rescale them with the raster map calculator
+[r.mapcalc](https://grass.osgeo.org/grass-stable/manuals/r.mapcalc.html).
+Write the expression `elevation = fractal / 10`
+to create a new elevation raster
+equal to the fractal surface divided by ten. 
+This is a local map algebra operation using simple arithmetic.
+Print the rescaled elevation range with 
+[r.info](https://grass.osgeo.org/grass-stable/manuals/r.info.html).
+
+::: {.panel-tabset group="language"}
+## Command line
+```{bash}
+r.mapcalc expression="elevation = fractal / 10"
+r.colors map=elevation color=elevation
+r.info -r map=elevation format=plain
+```
+## Python
+```{python}
+# Map algebra
+gs.mapcalc("elevation = fractal / 10")
+gs.run_command("r.colors", map="elevation", color="elevation")
+
+# Visualize
+m.d_rast(map="elevation")
+m.d_legend(raster="elevation", at=(5, 95, 1, 3))
+m.show()
+
+# Print range
+info = gs.read_command("r.info", map="elevation", flags="r")
+print(info)
+```
+:::
+
+![Mountains](images/map_algebra_05.webp)
+
+## Functions
+Map algebra has arithmetic functions as well as operators.
+Let's take the absolute value of our terrain 
+to model river valleys winding through our fractal mountains. 
+In the raster map calculator
+[r.mapcalc](https://grass.osgeo.org/grass-stable/manuals/r.mapcalc.html),
+write the expression `elevation = abs(elevation)`
+to create a new elevation raster
+equal to absolute value of the fractal surface.
+Because our fractal terrain has values
+both below and above mean sea level, i.e. zero, 
+taking its absolute value 
+will raise the low-lying valleys into mountains. 
+The elevation gradient close to zero will become river valleys
+at the feet of the mountains. 
+Use [r.colors](https://grass.osgeo.org/grass-stable/manuals/r.colors.html)
+to assign a color gradient to the elevation raster,
+ensuring that the gradient fits rescaled elevation range of the raster.
+
+::: {.panel-tabset group="language"}
+## Command line
+```{bash}
+r.mapcalc expression="elevation = abs(elevation)" --overwrite
+r.colors map=elevation color=elevation
+```
+## Python
+```{python}
+# Map algebra
+gs.mapcalc("elevation = abs(elevation)", overwrite=True)
+gs.run_command("r.colors", map="elevation", color="elevation")
+
+# Visualize
+m.d_rast(map="elevation")
+m.d_legend(raster="elevation", at=(5, 95, 1, 3))
+m.show()
+```
+:::
+
+![Valleys](images/map_algebra_06.webp)
+
+# Neighborhoods
```

Comment:
Perhaps add a note that you can use neighborhood operators in r.mapcalc itself, e.g. to design custom filters. And r.mfilter is another one.